homeassistant/components/miele/entity.py

```diff
@@ -0,0 +1,48 @@
+"""Entity base class for the Miele integration."""
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DEVICE_TYPE_TAGS, DOMAIN, MANUFACTURER, MieleAppliance, StateStatus
+from .coordinator import MieleDataUpdateCoordinator
+
+
+class MieleEntity(CoordinatorEntity[MieleDataUpdateCoordinator]):
+    """Base class for Miele entities."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: MieleDataUpdateCoordinator,
+        device_id: str,
+        description: EntityDescription,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self._device_id = device_id
+        self.entity_description = description
+        self._attr_unique_id = f"{device_id}-{description.key}"
+
+        device = self.coordinator.data.devices[self._device_id]
+        appliance_type = DEVICE_TYPE_TAGS.get(MieleAppliance(device.device_type))
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._device_id)},
+            serial_number=self._device_id,
+            name=appliance_type or device.tech_type,
+            translation_key=appliance_type,
+            manufacturer=MANUFACTURER,
+            model=device.tech_type,
+            hw_version=device.xkm_tech_type,
+            sw_version=device.xkm_release_version,
+        )
```

Comment:
```suggestion
        device = self.coordinator.data.devices[device_id]
        appliance_type = DEVICE_TYPE_TAGS.get(MieleAppliance(device.device_type))
        self._attr_device_info = DeviceInfo(
            identifiers={(DOMAIN, device_id)},
            serial_number=device_id,
            name=appliance_type or device.tech_type,
            translation_key=appliance_type,
            manufacturer=MANUFACTURER,
            model=device.tech_type,
            hw_version=device.xkm_tech_type,
            sw_version=device.xkm_release_version,
        )
```

Reply:
I took care of this and other changes last night but they disappeared into git stash for unknown reason.